dbt/include/trino/macros/materializations/incremental.sql

```diff
@@ -1,21 +1,77 @@
-{% macro dbt_trino_get_append_sql(tmp_relation, target_relation) %}
+{% macro validate_get_incremental_strategy(config) %}
+  {#-- Find and validate the incremental strategy #}
+  {%- set strategy = config.get("incremental_strategy", default="append") -%}
 
-    {%- set dest_columns = adapter.get_columns_in_relation(target_relation) -%}
+  {% set invalid_strategy_msg -%}
+    Invalid incremental strategy provided: {{ strategy }}
+    Expected one of: 'append', 'delete+insert'
+  {%- endset %}
+  {% if strategy not in ['append', 'delete+insert'] %}
+    {% do exceptions.raise_compiler_error(invalid_strategy_msg) %}
+  {% endif %}
+
+  {% do return(strategy) %}
+{% endmacro %}
+
+{% macro get_incremental_sql(strategy, tmp_relation, target_relation, unique_key, dest_columns) %}
+  {% if strategy == 'append' %}
+    {% do return(get_append_sql(target_relation, tmp_relation, dest_columns)) %}
+  {% elif strategy == 'delete+insert' %}
+    {% do return(get_delete_insert_merge_sql(target_relation, tmp_relation, unique_key, dest_columns)) %}
+  {% elif strategy == 'merge' %}
+    {#-- Not yet supported by trino (blocked in validate_get_incremental_strategy macro) --#}
+    {% do return(get_merge_sql(target_relation, tmp_relation, unique_key, dest_columns)) %}
+  {% endif %}
+{% endmacro %}
+
+{% macro get_append_sql(target_relation, tmp_relation, dest_columns) %}
```

Comment:
We've discussed this already.

I see `get_append_sql` and `trino__get_delete_insert_merge_sql` which are namewise not consistent.

I'd be in favor of prefixing with `trino__` until we get consistent macro names across all the codebase.

Reply:
I suggest that the `trino__` prefix should only be used for adapter macros that are dispatched in core dbt. The `get_append_sql` macro is not part of core dbt while `get_delete_insert_merge_sql` is. See https://github.com/dbt-labs/dbt-core/blob/1071a4681df91633301fdf23e34de819b66fbeb7/core/dbt/include/global_project/macros/materializations/models/incremental/merge.sql#L51-L53

Reply:
How do we know whether a certain macro is not going to be part of core dbt?

In any case keep this in mind for the eventual follow-up task of having consistent naming for the macros.